leases/schema/mutations.py

```diff
@@ -189,6 +189,71 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, old_lease_id, new_berth_id, **input):
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
+
+        old_lease = get_node_from_global_id(
+            info, old_lease_id, only_type=BerthLeaseNode, nullable=False,
+        )
```

Comment:
Should the old_lease be currently active? For example, prevent accidentally modifying leases from past seasons?
What about lease status? Is switching allowed for all lease statuses, including REFUSED/ERROR/TERMINATED/EXPIRED?

Reply:
Hmm, yea, probably a good idea to add a check if the lease is active.

At the moment `leases.schema.utils.terminate_lease` checks that the lease status is PAID `leases/schema/utils.py:89`:
```
    if lease.status != LeaseStatus.PAID:
        raise VenepaikkaGraphQLError(_(f"Lease is not paid: {lease.status}"))
```

Reply:
Added check for active leases.